### PR TITLE
Feature/racing

### PR DESCRIPTION
**Pull request by ksookdhew**
Feature/ Racing Screen
[Merge] Feature/Racing → Dev

 ## Description 
I added the Racing screen which displays all the races for the season. I also edited the race info screen to display more meaningful data.
 
## Screenshots/GIFs 
Screenshot 2024-04-30 at 08 59 57

 
## Checklist 
Please review and check the following before submitting this pull request: 
- [x] The code follows the project's coding standards. 
- [x] 100% Unit test coverage for View Models 
- [x] All warnings Resolved.

### CODE REVIEW

File: F1/F1/Race/RaceViewModel.swift [hunk 1]
```diff
@@ -64,6 +71,14 @@ class RaceViewModel {
 
     }
 
+    func race(atIndex: Int) -> RaceInfo {
+        allRaces[atIndex]
+    }
+
+    func imageName(circuitCode: String?) -> String {
+        return "\(circuitCode ?? "").png"
```
Comment (sjacobs1) [inline F1/F1/Race/RaceViewModel.swift:79]: would you be able to omit 'return' here?

File: F1/F1/Race/RacingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//
+
+import UIKit
+
+class RacingViewController: UIViewController {
+    private lazy var viewModel = RaceViewModel(repository: RaceRepository(), delegate: self)
+
+    // MARK: IBOutlets
+    @IBOutlet weak var collectionView: UICollectionView!
+
```
Comment (sjacobs1) [inline F1/F1/Race/RacingViewController.swift:15]: can it be private?

Comment (ksookdhew) [reply F1/F1/Race/RacingViewController.swift:15]: Nice catch

File: F1/F1/Race/RacingCollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+//
+//  Created by Kaitlyn Sookdhew on 2024/04/28.
+//
+
+import UIKit
+
+class RacingCollectionViewCell: UICollectionViewCell {
+
+    // MARK: IBOutlets
+    @IBOutlet weak var raceName: UILabel!
```
Comment (simekanidvt) [inline F1/F1/Race/RacingCollectionViewCell.swift]: Private

File: F1/F1/Race/RacingViewController.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }
+}
+
+// MARK: - ViewModelDelegate
+extension  RacingViewController: ViewModelDelegate {
+    func reloadView() {
+        collectionView.reloadData()
+    }
+
+    func show(error: String) {
```
Comment (simekanidvt) [inline F1/F1/Race/RacingViewController.swift:71]: Rather rename this function to something like showError(message: String...

**Comment by simekanidvt** [comment]
PR looks good. Well done

**Review by sjacobs1 — APPROVED** [review]

**Review by simekanidvt — APPROVED** [review]

**Pull request merged**